types/node/index.d.ts

```diff
@@ -748,6 +748,11 @@ declare module "http" {
     export interface ServerRequest extends IncomingMessage {
         connection: net.Socket;
     }
+
+    export interface ServerResponseHeaders {
+      [key: string]: number | string | string[];
```

Comment:
Are you sure that numbers are allowed here? I think header values are always strings (or array of strings in case the same header name is set several times).
In case numbers are really allowed I would expect that also an array of numbers is allowed.